### PR TITLE
Revise perfect hash to align with libgrape-lite's pthash

### LINKED ISSUE #1852: Use libgrape-lite's pthash to replace BBHash
Describe your problem
---------------------

Currently, the perfect hash map used by the vertex map has problems with the string type. Replace BBHash with pthash to align with libgrape-lite.

### PR DESCRIPTION
**Pull request by vegetableysm**
What do these changes do?
-------------------------

**OLD**:
Modifications of libgrape-lite perfect hash
- remove ```inline int8_t log2(size_t value)``` function from ```hashmap_indexer_impl.h```（Not used）
- remove ```namespace sync_comm``` and ```struct CommImpl``` from ```hashmap_indexer_impl.h```（Not used）
- remove ```Allocator<T>``` of ```std::vector<T, Allocator<T>> inner_``` in ```hashmap_indexer_impl.h```（No need）
- remove ```encode_vec``` ```encode_val``` and ```decode_val``` API from ```ref_vector.h``` (Not used)
- add ```struct murmurhasher``` to ```single_phf_view.h```  (reduce dependency)
- add ```struct external_mem_dumper``` to ```single_phf_view.h``` for dump data to blob (Reduce memcpy)
- remove ```std::set<size_t> idx``` in ```static void build(Iterator keys, uint64_t n, Dumper& dumper, int thread_num)``` of ```struct SinglePHFView``` in ```single_phf_view.h``` (No need)
- add ```static void build(Iterator keys, uint64_t n, pthash::single_phf<murmurhasher, pthash::dictionary_dictionary, true>& phf, int thread_num)``` API for ```struct SinglePHFView``` in ```single_phf_view.h``` （Reduce memcpy）
- remove ```void serialize(std::unique_ptr<IOADAPTOR_T>& writer)``` ```void deserialize(std::unique_ptr<IOADAPTOR_T>& reader)``` and ```void serialize_to_mem(std::vector<char>& buf)``` from ```class StringViewVector``` in ```string_view_vector.h``` (Not used)
- remove ```void Serialize(const std::string& path)``` and ```void Deserialize(const std::string& path)``` from ```class ImmPHIdxer``` in ```perfect_hash_indexer.h``` (Not used)
- overwrite ```finish()``` API of ```class PHIdxerViewBuilder``` in ```perfect_hash_indexer.h``` (For vineyard)
- replace ```std::vector<char> buffer_``` with ```std::shared_ptr<Blob> buffer_``` in ```perfect_hash_indexer.h``` (For vineyard)
- replace ```nonstd::string_view``` with ```arrow_string_view``` in all files. (For vineyard)

**NEW**:
- Use libgrapelite pthash
- Remove pthash / BBHash from vineyard

Related issue number
--------------------

Fixes #1852

### CODE REVIEW

File: modules/basic/ds/hashmap.h [hunk 1]
```diff
@@ -359,10 +362,11 @@ class PerfectHashmapBuilder : public PerfectHashmapBaseBuilder<K, V> {
     return Status::OK();
   }
 
-  boomphf::mphf<K, hasher_t> bphf_;
+  PHIdxerViewBuilder<K, uint64_t> builder_;
+  ImmPHIdxer<K, uint64_t> idxer_;
 
   const int concurrency_ = std::thread::hardware_concurrency();
-  const double gamma_ = 2.5f;
+  // const double gamma_ = 2.5f;
```
Comment (sighingnow) [inline modules/basic/ds/hashmap.h:369]: Remove if not used.

File: test/perfect_hashmap_test.cc [hunk 2]
```diff
@@ -21,24 +21,22 @@ limitations under the License.
...
-    return 1;
-  }
-  std::string ipc_socket = std::string(argv[1]);
-
+void test_double_id(std::string& ipc_socket) {
   Client client;
   VINEYARD_CHECK_OK(client.Connect(ipc_socket));
   LOG(INFO) << "Connected to IPCServer: " << ipc_socket;
 
+  // PerfectHashmapBuilder<int, double> builder(client);
```
Comment (sighingnow) [inline test/perfect_hashmap_test.cc:39]: Remove if not used.

Please go through the PR by yourself as a double check.

Thanks!

Comment (vegetableysm) [reply test/perfect_hashmap_test.cc:39]: ok

Comment (vegetableysm) [reply test/perfect_hashmap_test.cc:39]: I will check for all similar issues

**Comment by vegetableysm** [comment]
Refer to #1992

**Pull request closed without merging**